### PR TITLE
#160702928 place order endpoint

### PR DESCRIPTION
**Pull request by nwashangai**
#### What does this PR do?
it adds place order functionality
#### Description of Task to be completed?
* create an endpoint to place an order
* validate endpoint
* write test cases to place an order
#### How should this be manually tested?
* Not applicable
#### Any background context you want to provide?
* None
#### What are the relevant pivotal tracker stories?
#160702928
#### Screenshots (if appropriate)
* None
#### Questions:
* None

### CODE REVIEW

File: src/controllers/OrderController.js [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-          id: uuid(process.env.URL, uuid.URL),
-          dateTime: new Date(),
-          status: 'pending'
-        });
-        order.push(userOrder);
-        response.status(200).json({ status: 'success', message: 'Order placed', entry: request.body });
-    } else {
-      response.status(400).json({ status: 'error', message: 'invalid user ID' });
-    }
+    OrderModel.makeOrder(request.body).then((result) => {;
```
Comment (dogoyaro) [inline src/controllers/OrderController.js]: What happens when the order placement fails?

File: src/models/migration.js [hunk 2]
```diff
@@ -22,18 +22,40 @@ export default async () => {
         phone VARCHAR(15) not null,
         password TEXT not null
         )`
-      ).catch(error => { throw error });
+      ).then(async () => {
+        await query(`INSERT INTO users(
```
Comment (dogoyaro) [inline src/models/migration.js]: Why are you programmatically creating this user with this credentials here?

File: src/models/migration.js [hunk 3]
```diff
@@ -22,18 +22,40 @@ export default async () => {
...
+      });
   await query(`CREATE TABLE IF NOT EXISTS foods(
         id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
         name VARCHAR(100) not null,
         description TEXT not null,
         image TEXT, price MONEY not null
         )`
-      ).catch(error => { throw error });
+      ).then(async () => {
+        await query(`INSERT INTO foods(
```
Comment (dogoyaro) [inline src/models/migration.js]: Same for this. I don't think your migration files should insert any data into your database.

File: test/order.js [hunk 4]
```diff
@@ -8,86 +8,97 @@ chai.should();
...
 describe('Fast-Food-Fast orders test', () => {
   describe('Test endpoint to place order', () => {
-    it('it should place user\'s order when user provides order information', (done) => {
+    it('it should reject user\'s order when user has no access token', (done) => {
       chai.request(app)
         .post('/api/v1/orders')
         .send(order)
         .end((err, res) => {
-          res.should.have.status(200);
+          res.should.have.status(403);
```
Comment (dogoyaro) [inline test/order.js]: Should this be a 403?

File: src/models/migration.js (hunk 2 above)
Comment (nwashangai) [reply src/models/migration.js]: for test purposes

Comment (dogoyaro) [reply src/models/migration.js]: Don't do this here. Your tests should create the data themselves.

Comment (nwashangai) [reply src/models/migration.js]: but can I ignore the food menu creation until I create an endpoint for it because there is no endpoint for it?

Comment (dogoyaro) [reply src/models/migration.js]: I think you should do that.

File: src/models/migration.js (hunk 3 above)
Comment (nwashangai) [reply src/models/migration.js]: I thought you said you agreed I could leave this until the endpoint is created

**Pull request merged**